feat(sprint_3) sprint_3 complete

andreu6454 opened the pull request.

andreu6454 committed:
feat(*): start sprint_3

- Juiciest page done
- Recipe page done
- Error page done
- Carousel done
- Breadcrumbs done
- Navigation and Tabs done

andreu6454 committed:
feat(*): start sprint_3

- Juiciest page done
- Recipe page done
- Error page done
- Carousel done
- Breadcrumbs done
- Navigation and Tabs done

andreu6454 committed:
feat(*): add ErrorAlert

- ErrorAlert done
- AppLoader done

andreu6454 committed:
feat(*): fix BurgerMenu tests

- Burger menu done

andreu6454 committed:
feat(*): add response with params

- Filters done

andreu6454 committed:
feat(*): add response with params

- Filters done
- Search done
- Allergens done

andreu6454 committed:
refactor(*): small refactoring

- rename JuiciestFoodPage.tsx
- add infinity data logic to main page
- add errorsAlert to all get requests
- fix cardWithImage category badge

Ivan6813 commented:
# Результаты тестов
Процент пройденных тестов: 96%.
Общее количество тестов: 27.
Количество непройденных тестов: 1.
***
**Test cases for YeeDaa application -- рендер App компоненты -- Take a screenshot of the app components (failed).jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746806589/Test%20cases%20for%20YeeDaa%20application%20--%20рендер%20App%20компоненты%20--%20Take%20a%20screenshot%20of%20the%20app%20components%20(failed).jpg)
***
**allergens-1920.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746806589/allergens-1920.jpg)
***
**app-loader-1920.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746806589/app-loader-1920.jpg)
***
**app-loader-360.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746806589/app-loader-360.jpg)
***
**carousel-1920.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746806589/carousel-1920.jpg)
***
**carousel-360.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746806589/carousel-360.jpg)
***
**error-notification-360-not-visible.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746806589/error-notification-360-not-visible.jpg)
***
**error-notification-360.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746806589/error-notification-360.jpg)
***
**error-notification-768.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746806589/error-notification-768.jpg)
***
**error-page-360.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746806589/error-page-360.jpg)
***
**filter-before-clear-768.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746806589/filter-before-clear-768.jpg)
***
**filter-open-1920.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746806589/filter-open-1920.jpg)
***
**juiciest-page-1920.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746806589/juiciest-page-1920.jpg)
***
**juiciest-page-768-loading.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746806589/juiciest-page-768-loading.jpg)
***
**juiciest-page-768.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746806589/juiciest-page-768.jpg)
***
**open-drawer-360.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746806589/open-drawer-360.jpg)
***
**open-hamburger-360.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746806589/open-hamburger-360.jpg)
***
**open-hamburger-768.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746806589/open-hamburger-768.jpg)
***
**search-category-768.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746806589/search-category-768.jpg)
***
**search-loader-1440.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746806589/search-loader-1440.jpg)
***
**search-not-found-360.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746806589/search-not-found-360.jpg)
***
**Страница рецепта_1920x750 (1).jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746806589/Страница%20рецепта_1920x750%20(1).jpg)
***
**Страница рецепта_1920x750.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746806589/Страница%20рецепта_1920x750.jpg)
***
**Страница рецепта_360x1080.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746806589/Страница%20рецепта_360x1080.jpg)
***
**Страница рецепта_768x1080.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746806589/Страница%20рецепта_768x1080.jpg)

andreu6454 commented:
<img width="1440" alt="image" src="https://github.com/user-attachments/assets/476bfa34-f0f3-4f12-9b79-e79032cecde7" />

andreu6454 committed:
refactor(*): small refactoring

- remove loader from newRecipes
- add loader to RecipePage
- other small fixes

Ivan6813 commented:
# Результаты тестов
Процент пройденных тестов: 100%.
Общее количество тестов: 27.
Количество непройденных тестов: 0.
***
**allergens-1920.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746814069/allergens-1920.jpg)
***
**app-loader-1920.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746814069/app-loader-1920.jpg)
***
**app-loader-360.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746814069/app-loader-360.jpg)
***
**carousel-1920.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746814069/carousel-1920.jpg)
***
**carousel-360.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746814069/carousel-360.jpg)
***
**error-notification-360-not-visible.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746814069/error-notification-360-not-visible.jpg)
***
**error-notification-360.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746814069/error-notification-360.jpg)
***
**error-notification-768.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746814069/error-notification-768.jpg)
***
**error-page-360.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746814069/error-page-360.jpg)
***
**filter-before-clear-768.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746814069/filter-before-clear-768.jpg)
***
**filter-open-1920.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746814069/filter-open-1920.jpg)
***
**juiciest-page-1920.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746814069/juiciest-page-1920.jpg)
***
**juiciest-page-768-loading.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746814069/juiciest-page-768-loading.jpg)
***
**juiciest-page-768.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746814069/juiciest-page-768.jpg)
***
**open-drawer-360.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746814069/open-drawer-360.jpg)
***
**open-hamburger-360.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746814069/open-hamburger-360.jpg)
***
**open-hamburger-768.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746814069/open-hamburger-768.jpg)
***
**search-category-768.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746814069/search-category-768.jpg)
***
**search-loader-1440.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746814069/search-loader-1440.jpg)
***
**search-not-found-360.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746814069/search-not-found-360.jpg)
***
**Страница рецепта_1920x750 (1).jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746814069/Страница%20рецепта_1920x750%20(1).jpg)
***
**Страница рецепта_1920x750.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746814069/Страница%20рецепта_1920x750.jpg)
***
**Страница рецепта_360x1080.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746814069/Страница%20рецепта_360x1080.jpg)
***
**Страница рецепта_768x1080.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746814069/Страница%20рецепта_768x1080.jpg)

SergikEnergy commented on src/app/router/AppRouter.tsx:
```diff
@@ -1,17 +1,41 @@
-import { Route, Routes } from 'react-router';
```
![image](https://github.com/user-attachments/assets/edd469a1-6d5f-4ffc-a1a6-1b3f042e3e2c)
для загрузки добавить лоадер в кнопку и дизейбл сделать, чтобы избежать отработки лишнего онклика

andreu6454 replied:
там же тест вроде на слово 'загрузка'

andreu6454 replied:
<img width="394" alt="image" src="https://github.com/user-attachments/assets/cc22ef1b-e187-47bc-9350-701ecb68fd3d" />

SergikEnergy replied:
да - но никто ж не мешает справа или слева вместить в кнопку лоадер и ее сделать дизейблд

SergikEnergy commented on src/app/router/AppRouter.tsx:
```diff
@@ -1,17 +1,41 @@
-import { Route, Routes } from 'react-router';
+import { Navigate, Route, Routes } from 'react-router';
```
![image](https://github.com/user-attachments/assets/dbc44ae6-70ff-49c3-b99b-2e1336a9713a)
 если ничего не найдено, то тогда красный бордер у инпута

andreu6454 replied:
<img width="639" alt="image" src="https://github.com/user-attachments/assets/91705da6-bfc2-4249-a78a-7b5756905c67" />
хорошо, переделаю.
В фигме нет бордера, так и задумывалось?

SergikEnergy replied:
вспоминай прошлый спринт - все так же остается - просто текст добавился, дизайнер не перенес до конца

SergikEnergy commented on src/app/router/AppRouter.tsx:
```diff
@@ -1,17 +1,41 @@
-import { Route, Routes } from 'react-router';
+import { Navigate, Route, Routes } from 'react-router';
 
-import { JuiciestFood } from '~/pages/JuiciestFood/JuiciestFood';
```
![image](https://github.com/user-attachments/assets/691abafa-d385-446b-afd5-78d183549d4e)
на вот этих скринах должен быть алерт - его нет

andreu6454 replied:
<img width="1372" alt="image" src="https://github.com/user-attachments/assets/337111e4-cc8c-4772-b6b5-df92af4d67e3" />
На локальных тестах  в скриншотах есть, возможно до его появления был сделан скриншот

SergikEnergy replied:
ок

SergikEnergy commented on src/app/router/AppRouter.tsx:
```diff
@@ -1,17 +1,41 @@
+            {createRoutes({ data: categories })}
+            <Route path='/the-juiciest'>
+                <Route index element={<JuiciestFoodPage />} />
+                <Route path='/the-juiciest/:recipeId' element={<RecipePage />} />
```
такой путь не нужен - из сочного мы ведем на страницу рецепта по схеме категория/подкатегория/рецептИД

andreu6454 replied:
<img width="838" alt="image" src="https://github.com/user-attachments/assets/ce6e8e6d-b79e-4bde-8a03-b5f9c9a0ee1f" />
Исправлю, делал как в макете было.

SergikEnergy replied:
ну это немного ошибка, вроде в чатике обсуждали этот момент

SergikEnergy commented on src/app/router/AppRouter.tsx:
```diff
@@ -1,17 +1,41 @@
-    </Routes>
-);
+export const AppRouter = () => {
+    const { isLoading, isError } = useGetCategoriesQuery();
```
не нужно логики в роутеры впихывать - надо вынести в HOC - или еще как обыграть, здесь только чистые роуты

SergikEnergy commented on src/app/router/AppRouter.tsx:
```diff
@@ -1,17 +1,41 @@
+    const categories = useAppSelector(categoriesSelector);
+    const dispatch = useAppDispatch();
+
+    if (isLoading) {
+        return <FullScreenSpinner />;
+    }
```
спиннер тоже - лучше это все иметь на уровне Апп - или еще где

SergikEnergy commented on src/app/router/AppRouter.tsx:
```diff
@@ -1,17 +1,41 @@
+    return (
+        <Routes>
+            <Route index element={<MainPage />} />
+            {createRoutes({ data: categories })}
```
вот и логику тогда сюда вынести - или под это отдельный компонент-роут

SergikEnergy commented on src/app/router/routeComponents.tsx:
```diff
@@ -1,31 +1,40 @@
-        <Route path={`/${category.general}`} key={category.general}>
-            <Route index element={<Navigate to={category.links[0].link} replace />} />
+        <Route path={`/${category.category}`} key={category.category}>
+            <Route index element={<Navigate to={category.subCategories[0].category} replace />} />
 
-            {category.links.map((subItem) => (
-                <Route path={`${subItem.link}/`} key={subItem.link} element={<RecipesListPage />} />
+            {category.subCategories.map((subItem) => (
+                <Route
+                    path={`${subItem.category}/`}
+                    key={subItem.category}
+                    element={
```
почитай  как делать динамические параметры и используются через хук useParams
тогда роут пишется как :category/:subcategory/:recipeId - и вокруг этого обыгрываешь все

andreu6454 replied:
Я знаю про '/:category/:subcategory/:recipeId', но у меня специально так сделано, чтобы отсекать все лишние пути. Например  путь 'snacks/meat-snacks' пройдет, а 'snacks/meat-snackss' нет и будет редирект на 404.
Сейчас попробую через HOC  все реализовать.

SergikEnergy commented on src/app/router/routeComponents.tsx:
```diff
@@ -1,31 +1,40 @@
 
-            <Route path=':subcategory/:recipeId' element={<RecipePage />} />
+            <Route
+                path=':subcategory/:recipeId'
```
ну вот же) с категориями такая же песня

SergikEnergy commented on src/components/Breadcrumbs/Breadcrumbs.tsx:
```diff
@@ -28,11 +27,20 @@ export const Breadcrumbs: FC<SmartBreadcrumbsProps> = memo(
 
-            const isNumericPath = !isNaN(Number(path));
+            let displayName = pathNames.find((el) => el.category === path)?.title || path;
+            //  для все категорий и сабКатегорий
```
комментарии убираем

SergikEnergy commented on src/components/CardWithImage/CardWithImage.tsx:
```diff
@@ -97,45 +102,45 @@ export const CardWithImage = memo((props: CardWithImageProps) => {
+                        {recipe.title}
                     </Typography>
-                    {(size === 'Desktop' || size === 'Laptop') && (
+                    {(screenSize === 'Desktop' || screenSize === 'Laptop') && (
```
обычно используют чуть друго хук типа isTabletWidth - но если так хочешь - добавь тогда строки в константы - Screens допустим, чтобы не повторялось

andreu6454 replied:
Сам сделал isDesktopLaptop состояние для таких моментов, но тут забыл обновить.
Исправлю

SergikEnergy commented on src/components/CardWithLeftImage/CardWithLeftImage.tsx:
```diff
@@ -3,24 +3,20 @@ import { Button, Flex } from '@chakra-ui/react';
 import BookmarkIcon from '../../assets/svg/bookmark.svg';
 
 interface CardWithLeftImageProps {
-    size: 'Desktop' | 'Laptop' | 'Tablet' | 'Mobile';
```
вот про эти константы речь - или enum просится тут тогда

SergikEnergy commented on src/components/CardWithLeftImage/CardWithLeftImage.tsx:
```diff
@@ -85,71 +82,83 @@ export const CardWithLeftImage = memo((props: CardWithLeftImageProps) => {
             >
-                {!isSmall && (
+                {/*{!isSmall && (*/}
+                {/*    <CardBadge*/}
```
коментарии убираем везде по коду

SergikEnergy commented on src/components/FullScreenSpinner/FullScreenSpinner.tsx:
```diff
@@ -0,0 +1,38 @@
+
+export const FullScreenSpinner = () => {
+    useEffect(() => {
+        document.body.style.overflow = 'hidden';
+
+        return () => {
+            document.body.style.overflow = '';
```
лучше сначала сохрани prevOverflow=document.body.style.overflow
и уже в очистке к нему возвращайся - ведь там может быть и не пустая строка

aliona-kragel commented on src/components/CardWithImage/CardWithImage.tsx:
```diff
@@ -97,45 +102,45 @@ export const CardWithImage = memo((props: CardWithImageProps) => {
                 {/*    />*/}
                 {/*)}*/}
             </Box>
```
комменты удаляем

SergikEnergy commented on src/components/ScrollToTop/ScrollToTop.tsx:
```diff
@@ -2,20 +2,38 @@ import { useEffect } from 'react';
-            dispatch(setCurrentPageCategory([category, subcategory]));
+            dispatch(
+                setCurrentPageCategory(
+                    allCategories.find((el) => el.category === category) || undefined,
```
|| undefined лишнее - т.к. метод find как раз и возвращает undefined в случае чего

SergikEnergy commented on src/components/ScrollToTop/ScrollToTop.tsx:
```diff
@@ -2,20 +2,38 @@ import { useEffect } from 'react';
+            );
+            dispatch(
+                setCurrentPageSubCategory(
+                    allSubCategories.find((el) => el.category === subcategory) || undefined,
```
.

aliona-kragel commented on src/components/CardWithLeftImage/CardWithLeftImage.tsx:
```diff
@@ -85,71 +82,83 @@ export const CardWithLeftImage = memo((props: CardWithLeftImageProps) => {
+                {/*        name='Елена высоцкая'*/}
+                {/*    />*/}
+                {/*)}*/}
+                {isSmall && (
```
тут тоже чистим комменты

SergikEnergy commented on src/pages/ErrorPage/ErrorPage.tsx:
```diff
@@ -0,0 +1,100 @@
+
+import ErrorPageImage from '../../assets/errorPageImage.png';
+
+const sizes = {
```
nit: в чакре есть брейкпоинты - и ты их сам настраиваешь - и можешь сделать как раз под свои точки - чтобы не плодить везде такие объекты - на будущее на рефактор тебе

aliona-kragel commented on src/components/CardWithLeftImage/CardWithLeftImage.tsx:
```diff
@@ -85,71 +82,83 @@ export const CardWithLeftImage = memo((props: CardWithLeftImageProps) => {
+                                ) : (
+                                    part
+                                ),
+                            )}
```
предлагаю вынести в отдельную переменную recipe.title.split(...).map(...) выше по компоненту. Это улучшит читаемость JSX и упростит восприятие логики

SergikEnergy commented on src/pages/JuiciestFoodPage/JuiciestFoodPage.tsx:
```diff
@@ -0,0 +1,53 @@
+import { RecipesContainer } from '~/widgets/RecipesContainer/RecipesContainer';
+
+export const JuiciestFoodPage = memo(() => {
+    const [page, setPage] = useState<number>(1);
+    const [isLastPage, setIsLastPage] = useState<boolean>(false);
+    const [isButtonLoading, setIsButtonLoading] = useState<boolean>(false);
```
такие типы сам ТС выведет по дефолтному значению

SergikEnergy commented on src/pages/MainPage/JuciestFood/Juiciest.tsx:
```diff
@@ -17,39 +21,40 @@ const gap = {
-        })
-        .slice(0, 8);
+        const onClickHandler = () => {
+            navigate(`/the-juiciest/${recipe._id}`);
```
тут онКлик должен вести на категория/подкатегория/рецептИД

aliona-kragel commented on src/components/SearchBlock/SearchBlock.tsx:
```diff
@@ -1,83 +1,87 @@
+    //         onSearchHandleWithCategories();
+    //     }
+    // }, [allergens]);
+
```
и тут удаляем)

SergikEnergy commented on src/pages/MainPage/MainPage.tsx:
```diff
@@ -1,35 +1,85 @@
-                </Flex>
-            ) : (
+            <SearchBlock onSearchHandle={onSearchHandle} title='Приятного аппетита!' />
+            {allRecipes.length === 0 ? (
```
!allReceipts.length

SergikEnergy replied:
почему не поправил?)

SergikEnergy replied:
не вижу правок/реакции

SergikEnergy commented on src/pages/RecipePage/RecipePage.tsx:
```diff
@@ -26,12 +31,29 @@ const gaps = {
+    if (!data) return null;
+    if (isError) {
+        dispatch(setAppError('error'));
+        window.history.back();
```
тф==есть хук useNavigate - и там можно навиировать на "-1" - а то тут уже window.history

aliona-kragel commented on src/shared/types/recipeTypes.ts:
```diff
@@ -0,0 +1,38 @@
+    bookmarks: number;
+    createdAt: string;
+    _id: string;
+}
```
опциональные в конец и лучше использовать type

SergikEnergy commented on src/query/services/recipes.ts:
```diff
@@ -0,0 +1,126 @@
+                        sortOrder: 'desc',
+                        page: page,
+                        limit: 8,
+                    },
```
используются одни и теже параметры - соответственно нужен просто 1 ендпоинт с обобщенными параметрами - а ты уже сам внутрь хука их пробрасываешь, какие тебе надо

SergikEnergy replied:
нет изменений здесь тоже

aliona-kragel commented on src/store/recipeListFilterService/recipeListFilterService.tsx:
```diff
@@ -1,30 +1,31 @@
+//
+//         return matchesCategory && matchesSubcategory;
+//     });
+// };
```
комменты

SergikEnergy commented on src/shared/hooks/useCategoryById.ts:
```diff
@@ -0,0 +1,12 @@
+import { categoriesSelector, subCategoriesSelector } from '~/store/categories-slice';
+import { useAppSelector } from '~/store/hooks';
+
+export const useCategoryById = (categoryId: string, subCategoryId: string) => {
```
опять же из useParams все это можно получить все ИДшки прям здесь внутри хука

SergikEnergy commented on src/shared/hooks/useGetQueryParams.ts:
```diff
@@ -0,0 +1,48 @@
+            queryParams.allergens = allergens.join(',').replace(/\s/g, '').toLowerCase();
+        }
+        if (searchString) queryParams.searchString = searchString.toLowerCase();
+        if (meat.length > 0) {
```
if (meat.length > 0) {   => if(meat.length)

SergikEnergy reviewed (changes requested):
смотри комментарии

aliona-kragel reviewed (changes requested):
небольшие правки сделай

andreu6454 committed:
refactor(*): fix review changes

- remove logic from appRouter
- remove comments
- fix onclick on juiciestCard
- remove redundant types

Ivan6813 commented:
# Результаты тестов
Процент пройденных тестов: 100%.
Общее количество тестов: 27.
Количество непройденных тестов: 0.
***
**allergens-1920.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746830881/allergens-1920.jpg)
***
**app-loader-1920.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746830881/app-loader-1920.jpg)
***
**app-loader-360.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746830881/app-loader-360.jpg)
***
**carousel-1920.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746830881/carousel-1920.jpg)
***
**carousel-360.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746830881/carousel-360.jpg)
***
**error-notification-360-not-visible.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746830881/error-notification-360-not-visible.jpg)
***
**error-notification-360.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746830881/error-notification-360.jpg)
***
**error-notification-768.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746830881/error-notification-768.jpg)
***
**error-page-360.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746830881/error-page-360.jpg)
***
**filter-before-clear-768.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746830881/filter-before-clear-768.jpg)
***
**filter-open-1920.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746830881/filter-open-1920.jpg)
***
**juiciest-page-1920.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746830881/juiciest-page-1920.jpg)
***
**juiciest-page-768-loading.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746830881/juiciest-page-768-loading.jpg)
***
**juiciest-page-768.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746830881/juiciest-page-768.jpg)
***
**open-drawer-360.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746830881/open-drawer-360.jpg)
***
**open-hamburger-360.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746830881/open-hamburger-360.jpg)
***
**open-hamburger-768.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746830881/open-hamburger-768.jpg)
***
**search-category-768.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746830881/search-category-768.jpg)
***
**search-loader-1440.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746830881/search-loader-1440.jpg)
***
**search-not-found-360.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746830881/search-not-found-360.jpg)
***
**Страница рецепта_1920x750 (1).jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746830881/Страница%20рецепта_1920x750%20(1).jpg)
***
**Страница рецепта_1920x750.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746830881/Страница%20рецепта_1920x750.jpg)
***
**Страница рецепта_360x1080.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746830881/Страница%20рецепта_360x1080.jpg)
***
**Страница рецепта_768x1080.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746830881/Страница%20рецепта_768x1080.jpg)

SergikEnergy reviewed (changes requested):
не вижу правок остальных или коментов от тебя, пока жду еще

andreu6454 commented:
Поправлю через 30 мин. Хотел изменить, пропустил.

andreu6454 committed:
refactor(*): fix review changes

- remove redundant api endpoints
- other small fixes

Ivan6813 commented:
# Результаты тестов
Процент пройденных тестов: 100%.
Общее количество тестов: 27.
Количество непройденных тестов: 0.
***
**allergens-1920.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746837131/allergens-1920.jpg)
***
**app-loader-1920.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746837131/app-loader-1920.jpg)
***
**app-loader-360.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746837131/app-loader-360.jpg)
***
**carousel-1920.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746837131/carousel-1920.jpg)
***
**carousel-360.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746837131/carousel-360.jpg)
***
**error-notification-360-not-visible.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746837131/error-notification-360-not-visible.jpg)
***
**error-notification-360.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746837131/error-notification-360.jpg)
***
**error-notification-768.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746837131/error-notification-768.jpg)
***
**error-page-360.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746837131/error-page-360.jpg)
***
**filter-before-clear-768.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746837131/filter-before-clear-768.jpg)
***
**filter-open-1920.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746837131/filter-open-1920.jpg)
***
**juiciest-page-1920.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746837131/juiciest-page-1920.jpg)
***
**juiciest-page-768-loading.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746837131/juiciest-page-768-loading.jpg)
***
**juiciest-page-768.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746837131/juiciest-page-768.jpg)
***
**open-drawer-360.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746837131/open-drawer-360.jpg)
***
**open-hamburger-360.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746837131/open-hamburger-360.jpg)
***
**open-hamburger-768.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746837131/open-hamburger-768.jpg)
***
**search-category-768.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746837131/search-category-768.jpg)
***
**search-loader-1440.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746837131/search-loader-1440.jpg)
***
**search-not-found-360.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746837131/search-not-found-360.jpg)
***
**Страница рецепта_1920x750 (1).jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746837131/Страница%20рецепта_1920x750%20(1).jpg)
***
**Страница рецепта_1920x750.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746837131/Страница%20рецепта_1920x750.jpg)
***
**Страница рецепта_360x1080.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746837131/Страница%20рецепта_360x1080.jpg)
***
**Страница рецепта_768x1080.jpg**
![Скриншот автотестов](https://static.clevertec.ru/media/sprint1/andreu6454/1746837131/Страница%20рецепта_768x1080.jpg)

andreu6454 commented:
'getRecipeByCategory' оставлил для тестов
'getRecipesWithParams' оставил чтоб сильно логику mainPage не менять 
остальные удалил
